src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.cs

```diff
@@ -100,14 +107,28 @@ async ValueTask<VersionStamp> GetTextVersionCoreAsync(CancellationToken cancella
         }
     }
 
-    public DocumentState UpdateVersion()
-        => new(this, _textAndVersionSource);
+    public bool TryGetCodeDocument([NotNullWhen(true)] out RazorCodeDocument? result)
+        => _generatedOutputSource.TryGetValue(out result);
+
+    public ValueTask<RazorCodeDocument> GetCodeDocumentAsync(DocumentSnapshot document, CancellationToken cancellationToken)
+        => _generatedOutputSource.GetValueAsync(document, cancellationToken);
+
+    public DocumentState UpdateImport()
+        // Be a bit optimistic and try to reuse the output.
+        => new(this, _textAndVersionSource, _generatedOutputSource.ForkIfOutputAvailable());
```

Comment:
This one scares me. If you tell me this is how it used to work, then fine, but if a using directive is added to an `_Imports.razor` file, then all documents that import that file _will_ get a new using directive in their compiled output.

Do we understand how things "will catch up eventually"? If that is well understood, and we know it will always eventually happen then fine, but I'm a little nervous.

Again though, if this is how it used to work, my concerns aren't blocking. I'm nervous, but thats not to think that I wasn't nervous about the old code either. At least you understand this code :)

Reply:
Yeah, I don't love this change and am going to close the PR. I was hoping to see positive results in Speedometer, but no improvements manifested.

Reply:
> Again though, if this is how it used to work, my concerns aren't blocking. I'm nervous, but thats not to think that I wasn't nervous about the old code either. At least you understand this code :)

I appreciate you for sharing your concern.